Task 2 user crud

ronaldhej opened the pull request:
- Added CRUD for users for admins
- Added ability to create a user with a certain role
- Added role enum to utils
- Added DTOs for user crud operations

LokeSGJ commented on GirafAPI/Endpoints/UserEndpoints.cs:
```diff
@@ -1,32 +1,93 @@
+                        break;
+                }
+
+                Results.Created($"/users/{user.Id}", user);
```
Does this not need to be a return?

LokeSGJ commented on GirafAPI/Endpoints/UserEndpoints.cs:
```diff
@@ -1,32 +1,93 @@
 
-            // LATER IMPLEMENTATION assign roles
-            // await userManager.AddToRoleAsync(user, "RoleName");
+            Results.BadRequest(result.Errors);
```
Does this not also need to be a return?

LokeSGJ reviewed (changes requested):
Looks good overall. Can you please move the Entities/DTOs directory with the user DTOs to Entities/Users/DTOs?

ronaldhej commented:
I moved the requested files and added explicit returns to the Results just in case, I think it auto returns the last line though. Fixed the merge conflicts, one of the conflicting files is auto generated though so it should correctly reflect the model in the main branch anyway
